File: p2-back/src/main/java/br/fatec/p2Cloud/model/Cliente.java
package br.fatec.p2Cloud.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;

@Entity
@Table(name = "clientes")
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nome;

    @Column(unique = true)
    @Pattern(regexp = "^\\d{3}\\.?\\d{3}\\.?\\d{3}-?\\d{2}$", message = "CPF inválido")
    private String cpf;

    private String telefone;

    @Email(message = "Email inválido")
    private String email;

    private String endereco;

    @ManyToOne
    @JoinColumn(name = "barbearia_id")
    @JsonIgnoreProperties({"clientes", "agendas", "servicos"})
    private Barbearia barbearia;

    // Construtores
    public Cliente() {
    }

    public Cliente(String nome, String cpf, String telefone, String email, String endereco) {
        this.nome = nome;
        this.cpf = cpf;
        this.telefone = telefone;
        this.email = email;
        this.endereco = endereco;
    }

    // Getters e Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public String getTelefone() {
        return telefone;
    }

    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getEndereco() {
        return endereco;
    }

    public void setEndereco(String endereco) {
        this.endereco = endereco;
    }

    public Barbearia getBarbearia() {
        return barbearia;
    }

    public void setBarbearia(Barbearia barbearia) {
        this.barbearia = barbearia;
    }
}

File: p2-front/src/models/cliente.js
// Model da classe Cliente

/**
 * Representa um cliente da barbearia.
 */
export default class Cliente {
  /**
   * @param {Object} options
   * @param {string} options.nome
   * @param {string} options.cpf
   * @param {string} options.telefone
   * @param {string} options.email
   * @param {string|Object} options.endereco
   */
  constructor({ nome = '', cpf = '', telefone = '', email = '', endereco = '' } = {}) {
    this._nome = nome;
    this._cpf = cpf;
    this._telefone = telefone;
    this._email = email;
    this._endereco = endereco;
  }

  get nome() {
    return this._nome;
  }

  set nome(value) {
    this._nome = value;
  }

  get cpf() {
    return this._cpf;
  }

  set cpf(value) {
    this._cpf = value;
  }

  get telefone() {
    return this._telefone;
  }

  set telefone(value) {
    this._telefone = value;
  }

  get email() {
    return this._email;
  }

  set email(value) {
    this._email = value;
  }

  get endereco() {
    return this._endereco;
  }

  set endereco(value) {
    this._endereco = value;
  }

  toJSON() {
    return {
      nome: this._nome,
      cpf: this._cpf,
      telefone: this._telefone,
      email: this._email,
      endereco: this._endereco,
    };
  }
}
